packages/common-web/src/tid.ts

```diff
@@ -36,11 +36,15 @@ export class TID {
     if (clockid === null) {
       clockid = Math.floor(Math.random() * 32)
     }
-    return TID.fromTime(timestamp, clockid)
+    const tid = TID.fromTime(timestamp, clockid)
+    if (!prev || tid.newerThan(prev)) {
+      return tid
+    }
+    return TID.fromTime(prev.timestamp() + 1, clockid)
```

Comment:
I think we also have some leftover bits in `clockid` if that's useful.  One question I have is—would we take the same approach here of incrementing the timestamp if our timestamps already had microsecond precision?

Reply:
Yea I think we would - likelihood is not that timestamps collide but one has a lower clockid. likelihood is some clockdrift so we'll want to increment the timestamp itself

the clockid is actually just mean to for a multiwriter world to prevent id collision (inspired by twitter snowflakes)

the tid (in JS at least) is actually make of three parts

unix epoch millisecond + count to prevent millisecond collisions + clockid (unique value for machine generating TIDs)